pulsar-functions/runtime/src/main/java/org/apache/pulsar/functions/runtime/RuntimeUtils.java

```diff
@@ -221,8 +219,8 @@
             goInstanceConfig.setMaxMessageRetries(instanceConfig.getFunctionDetails().getRetryDetails().getMaxMessageRetries());
         }
 
-        if (metricsPort > 0 && metricsPort < 65536) {
-            goInstanceConfig.setMetricsPort(metricsPort);
+        if (instanceConfig.getMetricsPort() > 0 && instanceConfig.getMetricsPort() < 65536) {
```

Comment:
Can we use the `instanceConfig. hasValidMetricsPort()`?

Reply:
sure, forget to reuse it. ;P